test/geocel/GenericGeoTestInterface.hh

```diff
@@ -32,54 +34,80 @@ struct GenericGeoVolumeStackResult;
 /*!
  * Access capabilities from any templated GenericGeo test.
  *
- * The volume/instance offsets are usually used with Geant4, which has volume
- * IDs that may not start with zero if the problem has been reinitaialized.
- * (This is because geant4 uses global static integers for counting.) It can
- * also be used in other circumstances (vecgeom internal construction) where
- * fake volumes/instances are inserted before the "real" volumes/instances.
+ * \todo This is being refactored into GenericGeoTestBase so that we can just
+ * use the GeoTrackInterface and GeoParamsInterface wrappers.
  */
-class GenericGeoTestInterface
+class GenericGeoTestInterface : public LazyGeantGeoManager
 {
   public:
     //!@{
     //! \name Type aliases
     using TrackingResult = GenericGeoTrackingResult;
     using VolumeStackResult = GenericGeoVolumeStackResult;
+    using GeoTrackView = GeoTrackInterface<real_type>;
+    using UPGeoTrack = std::unique_ptr<GeoTrackView>;
     //!@}
 
   public:
-    //! Generate a track
-    virtual TrackingResult track(Real3 const& pos_cm, Real3 const& dir) = 0;
+    //// TESTS ////
 
-    //! Get the safety tolerance (defaults to SoftEq tol) for tracking result
-    virtual GenericGeoTrackingTolerance tracking_tol() const;
+    // Track until exiting the geometry
+    TrackingResult track(Real3 const& pos_cm, Real3 const& dir);
+    // Obtain the "touchable history" at a point
+    VolumeStackResult volume_stack(Real3 const& pos_cm);
```

Comment:
These used to be in the derived class but since they're no longer templated (they call the wrapped "make geo track view") I moved them here.